### PR TITLE
sdit-644 switch incentive migration to use generic code

### PR DESCRIPTION
**Pull request by lbennett-moj**

### CODE REVIEW

File: src/main/kotlin/uk/gov/justice/digital/hmpps/prisonerfromnomismigration/incentives/IncentivesMigrationService.kt [hunk 1]
```diff
@@ -274,40 +165,16 @@ class IncentivesMigrationService(
       e
     )
     queueService.sendMessage(
-      RETRY_INCENTIVE_MAPPING,
+      MigrationMessageType.RETRY_MIGRATION_MAPPING,
       MigrationContext(
```
Comment (lbennett-moj) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/prisonerfromnomismigration/incentives/IncentivesMigrationService.kt:169]: I'll tidy up the import of enums in next PR

File: src/main/kotlin/uk/gov/justice/digital/hmpps/prisonerfromnomismigration/incentives/IncentivesPrisonOffenderEventListener.kt [hunk 2]
```diff
@@ -94,3 +95,7 @@ data class IncentiveDeletedOffenderEvent(val offenderIdDisplay: String, val book
 data class IncentiveBooking(
   val nomisBookingId: Long
 )
+
+enum class IncentiveSynchronisationMessageType {
+  SYNCHRONISE_CURRENT_INCENTIVE
+}
```
Comment (lbennett-moj) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/prisonerfromnomismigration/incentives/IncentivesPrisonOffenderEventListener.kt:101]: todo find a nice way to extend enums

File: src/test/kotlin/uk/gov/justice/digital/hmpps/prisonerfromnomismigration/incentives/IncentivesMigrationIntTest.kt [hunk 3]
```diff
@@ -100,7 +100,7 @@ class IncentivesMigrationIntTest : SqsIntegrationTestBase() {
 
       await atMost Duration.ofSeconds(60) untilAsserted {
         verify(telemetryClient).trackEvent(
-          eq("nomis-migration-incentives-completed"),
+          eq("nomis-migration-completed"),
           any(),
```
Comment (lbennett-moj) [inline src/test/kotlin/uk/gov/justice/digital/hmpps/prisonerfromnomismigration/incentives/IncentivesMigrationIntTest.kt:104]: improve custom event names and switch to reflection approach for bean properties for telemetry

File: src/main/kotlin/uk/gov/justice/digital/hmpps/prisonerfromnomismigration/incentives/IncentiveMappingService.kt [hunk 4]
```diff
@@ -48,15 +48,13 @@ class IncentiveMappingService(@Qualifier("mappingApiWebClient") private val webC
...
     createNomisIncentiveMapping(
-      nomisBookingId = nomisBookingId,
-      nomisIncentiveSequence = nomisIncentiveSequence,
-      incentiveId = incentiveId,
-      mappingType = "NOMIS_CREATED"
+      nomisBookingId = incentiveNomisMapping.nomisBookingId,
+      nomisIncentiveSequence = incentiveNomisMapping.nomisIncentiveSequence,
+      incentiveId = incentiveNomisMapping.incentiveId,
+      mappingType = incentiveNomisMapping.mappingType
     )
```
Comment (lbennett-moj) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/prisonerfromnomismigration/incentives/IncentiveMappingService.kt:58]: will combine the synch and migration versions of these functions

**Review by andymarke — APPROVED** [review]
LGTM

**Pull request merged**